### PR TITLE
[DIA-1100] Add/Delete property buttons android tv

### PR DESCRIPTION
**Pull request by Nevazhnovu**

### CODE REVIEW

File: samples/metaapp/src/main/java/com/sourcepointmeta/metaapp/tv/properties/PropertyListFragmentTv.kt [hunk 1]
```diff
@@ -53,6 +52,19 @@ class PropertyListFragmentTv : VerticalGridSupportFragment(), OnItemViewClickedL
         requireContext().showPropertyDetail(propDto.propertyName)
     }
 
+    private fun titleInit(){
```
Comment (carmelo-iriti) [inline samples/metaapp/src/main/java/com/sourcepointmeta/metaapp/tv/properties/PropertyListFragmentTv.kt:55]: this method can be turned in an extension

File: samples/metaapp/src/main/java/com/sourcepointmeta/metaapp/tv/properties/PropertyListTitleView.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import android.graphics.drawable.Drawable
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.*
+import androidx.leanback.widget.TitleViewAdapter
+import com.sourcepointmeta.metaapp.R
+
+
+class PropertyListTitleView(context: Context?, attrs: AttributeSet?, defStyle: Int) : RelativeLayout(context, attrs, defStyle), TitleViewAdapter.Provider {
```
Comment (carmelo-iriti) [inline samples/metaapp/src/main/java/com/sourcepointmeta/metaapp/tv/properties/PropertyListTitleView.kt:14]: 1. extend the ConstraintLayout
2. Remove the variable declared programmatically from here (it is not possible to edit the layout from the xml)
3. The adapter needs a dedicated class
4. Use only one listener sending the action type as an enum o int and use a property instead a set method

File: samples/metaapp/src/main/res/layout-television/property_list_title.xml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+<androidx.constraintlayout.widget.ConstraintLayout
```
Comment (carmelo-iriti) [inline samples/metaapp/src/main/res/layout-television/property_list_title.xml:1]: merge the property_list_title.xml with property_list_title_leanback.xml in only one xml (https://github.com/SourcePointUSA/android-cmp-app/blob/develop/samples/metaapp/src/main/res/layout-television/property_item.xml)

File: samples/metaapp/src/main/java/com/sourcepointmeta/metaapp/tv/properties/PropertyListFragmentTv.kt (hunk 1 above)
Comment (Nevazhnovu) [reply samples/metaapp/src/main/java/com/sourcepointmeta/metaapp/tv/properties/PropertyListFragmentTv.kt]: @carmelo-iriti done

File: samples/metaapp/src/main/java/com/sourcepointmeta/metaapp/tv/properties/PropertyListTitleView.kt (hunk 2 above)
Comment (Nevazhnovu) [reply samples/metaapp/src/main/java/com/sourcepointmeta/metaapp/tv/properties/PropertyListTitleView.kt]: @carmelo-iriti done

File: samples/metaapp/src/main/java/com/sourcepointmeta/metaapp/tv/properties/PropertyListTitleView.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    constructor(context: Context) : this(context, null) {}
+    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0) {}
+
+    override fun getTitleViewAdapter(): TitleViewAdapter {
+        return mTitleViewAdapter
+    }
+
+    init {
+        val root: View = LayoutInflater.from(context).inflate(R.layout.property_list_title, this, true)
```
Comment (carmelo-iriti) [inline samples/metaapp/src/main/java/com/sourcepointmeta/metaapp/tv/properties/PropertyListTitleView.kt:30]: no need to inflate from here

File: samples/metaapp/src/main/java/com/sourcepointmeta/metaapp/tv/properties/PropertyListTitleView.kt [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+        val mRemoveAllBtn: Button? = root.findViewById(R.id.removeAllBtn)
```
Comment (carmelo-iriti) [inline samples/metaapp/src/main/java/com/sourcepointmeta/metaapp/tv/properties/PropertyListTitleView.kt:32]: no need to set the listener here

File: samples/metaapp/src/main/res/layout-television/property_list_title.xml (hunk 3 above)
Comment (carmelo-iriti) [reply samples/metaapp/src/main/res/layout-television/property_list_title.xml:1]: replace 
`androidx.constraintlayout.widget.ConstraintLayout`
with
`com.sourcepointmeta.metaapp.tv.properties.PropertyListTitleView`

File: samples/metaapp/src/main/java/com/sourcepointmeta/metaapp/tv/properties/PropertyListTitleView.kt (hunk 4 above)
Comment (Nevazhnovu) [reply samples/metaapp/src/main/java/com/sourcepointmeta/metaapp/tv/properties/PropertyListTitleView.kt:30]: @carmelo-iriti could you please pinpoint where I should inflate and init listeners? I thought that the best place is here

**Comment by carmelo-iriti** [comment]
Something is wrong in the UI 
Here there are the screenshots  https://drive.google.com/drive/folders/1kOKzk6oXrT8MmEI2yK5Hmsds-bNABNIi?usp=sharing
1 .Looks like that there are 2 + buttons in the main screen and also in the summary screen 
2. The Title "Metaapp 6.7.X" is not visible
3. The clear button is not visible

File: samples/metaapp/src/main/res/layout-television/property_list_title.xml [hunk 6]
```diff
@@ -55,6 +42,16 @@
...
         app:layout_constraintTop_toTopOf="parent" />
 
+    <TextView
+        android:id="@+id/titleTextView"
+        android:layout_width="wrap_content"
+        android:layout_height="64dp"
+        android:layout_centerVertical="true"
+        android:focusable="false"
+        android:gravity="center"
+        android:textAppearance="@android:style/TextAppearance.Large"
```
Comment (carmelo-iriti) [inline samples/metaapp/src/main/res/layout-television/property_list_title.xml:52]: use the following style for the title 
https://github.com/SourcePointUSA/android-cmp-app/blob/1c034b128d7c176e42f1684bac61a3bf62787374/samples/metaapp/src/main/res/layout-television/demo_header.xml#L24

File: samples/metaapp/src/main/res/layout-television/property_list_title.xml [hunk 7]
```diff
@@ -12,19 +12,6 @@
...
-        android:layout_height="80dp"
-        android:layout_marginEnd="24dp"
-        android:layout_centerVertical="true"
-        android:focusable="false"
-        android:layout_margin="6dp"
-        app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
-
     <ImageButton
         android:id="@+id/removeAllBtn"
```
Comment (carmelo-iriti) [inline samples/metaapp/src/main/res/layout-television/property_list_title.xml:16]: From the style point of view, it looks better switching the position of the 2 buttons:
+ btn first and cancel btn second

**Comment by carmelo-iriti** [comment]
We still have the Add and Remove btns in the summary screen https://drive.google.com/file/d/1wOXBLmtgnVvVXdH68nK8LSE0CgFn6ST8/view?usp=sharing
We don’t need the btns there

File: samples/metaapp/src/main/res/layout-television/property_list_title.xml [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        android:clickable="true"
+        android:clipToPadding="false"
+        android:focusable="true"
+        android:src="@android:drawable/ic_menu_delete"
+        android:stateListAnimator="@xml/animate_scale"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <ImageButton
+        android:id="@+id/addBtn"
```
Comment (carmelo-iriti) [inline samples/metaapp/src/main/res/layout-television/property_list_title.xml:31]: please use the underscore notation in the xml files (like browse_title_group)

File: samples/metaapp/src/main/res/layout-television/property_list_title.xml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        android:clickable="true"
+        android:clipToPadding="false"
+        android:focusable="true"
+        android:src="@drawable/add_selector"
+        android:stateListAnimator="@xml/animate_scale"
+        app:layout_constraintStart_toEndOf="@+id/removeAllBtn"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <TextView
+        android:id="@+id/titleTextView"
```
Comment (carmelo-iriti) [inline samples/metaapp/src/main/res/layout-television/property_list_title.xml:46]: pls use the underscore notation

**Comment by Nevazhnovu** [comment]
@carmelo-iriti done, check it out!

**Review by carmelo-iriti — APPROVED** [review]

**Pull request merged**